src/keyring/ed25519/signing_key.rs

```diff
@@ -1,75 +1,149 @@
 use super::{Signature, VerifyingKey};
 use crate::error::{Error, ErrorKind};
+use ed25519_dalek::hazmat::{raw_sign, ExpandedSecretKey};
 use signature::Signer;
 
 /// Signing key serialized as bytes.
 type SigningKeyBytes = [u8; SigningKey::BYTE_SIZE];
 
 /// Ed25519 signing key.
-#[derive(Clone, Debug)]
-pub struct SigningKey(ed25519_consensus::SigningKey);
+#[allow(clippy::large_enum_variant)]
+#[derive(Debug)]
+pub enum SigningKey {
+    /// Ed25519 signing key.
+    Ed25519(ed25519_consensus::SigningKey),
+    /// Ed25519 expanded signing key.
+    Ed25519Expanded(ExpandedSecretKey),
+}
```

Comment:
If we're going this route, I think I'd like to just migrate wholesale to `ExpandedSecretKey` as the internal type, rather than using an enum